test/models/user_test.exs

```diff
@@ -0,0 +1,86 @@
+defmodule Yggdrasil.UserTest do
+  use Yggdrasil.ModelCase
+
+  alias Yggdrasil.User
+
+  @valid_attrs %{username: "tester", password: "password", password_confirmation: "password"}
+  @valid_attrs_upcase_user %{username: "TESTER", password: "password", password_confirmation: "password"}
+
+  @missing_username %{password: "password", password_confirmation: "pass"}
+  @missing_password %{username: "tester", password_confirmation: "password"}
+  @missing_password_conf %{username: "tester", password: "password"}
+  @password_mismatch %{username: "tester", password: "test", password_confirmation: "tester"}
+  @password_bad_len %{username: "tester", password: "te", password_confirmation: "tester"}
+  @password_conf_bad_len %{username: "tester", password: "te", password_confirmation: "te"}
+
+  @blank_msg "can't be blank"
+  @doesntmatch_msg "does not match confirmation"
+  @unique_msg "has already been taken"
+
+  @invalid_length_msg "should be at least %{count} character(s)"
+  @min_count 4
+
+  test "changeset with valid attributes" do
+    changeset = User.changeset(%User{}, @valid_attrs)
+    assert changeset.valid?
+  end
+
+  test "changeset with missing username" do
+    changeset = User.changeset(%User{}, @missing_username)
+
+    assert changeset.errors[:username] == @blank_msg
+    refute changeset.valid?
+  end
+
+  test "changeset with missing password" do
+    changeset = User.changeset(%User{}, @missing_password)
+
+    assert changeset.errors[:password] == @blank_msg
+    refute changeset.valid?
```

Comment:
Right, noticed that as well also if you read the comment above about password being included in the changeset I mention that it probably should be split out. 

create_changeset
update_changeset

or something like that. I'm not sure we would want to call it create_user, as it's not doing anything other than producing a changeset to use to insert the user.